src/display_gadfly.jl

```diff
@@ -190,18 +190,9 @@ function dropfig(d::GadflyDisplay, i::Int)
     d.current_fig = isempty(d.fig_order) ? 0 : d.fig_order[end]
 end
 
-"""
-`figure(;name="Figure \$n", width=400, height=400)` creates a new
-figure window for displaying plots.
 
-`figure(n)` raises the `n`th figure window and makes it the current
-default plotting window, and returns the
-"""
-function figure(;name::String="Figure $(nextfig(_display))",
-                 width::Integer=400,    # TODO: make configurable
-                 height::Integer=400)
-    i = nextfig(_display)
-    c = gtkwindow(name, width, height, (x...)->dropfig(_display,i))
+
+function createFigure(c::GtkCanvas)
```

Comment:
How about calling this `initialize_toolbar_callbacks` or something? It could take a `Figure` input, and then you have a convenience method that creates a `Figure` from a `Canvas` before calling it.Or (since that's just one line) you could make the caller do that part.